### DISCUSSION TITLE
Why send multiple textDocument/didOpen in websocket when has multiple monaco-editor in page.

### DISCUSSION THREAD

**Discussion by KamiC6238** [TypeFox/monaco-languageclient#453] [Q&A]
my initialization code
![image](https://user-images.githubusercontent.com/23523595/212624398-449a0963-ecf0-4f86-990f-7627f43c8955.png)

in websocket
![image](https://user-images.githubusercontent.com/23523595/212624722-7b2c57f5-af70-45fc-b6e7-95d92d140051.png)

I would like to ask why didOpen is sent twice in a websocket connection. thanks so much if someone could tell me how to solve this problem.

version:
monaco-editor: 0.34.0
monaco-languageclient: 4.0.0

**Comment by CGNonofr** [comment]
Hello

It's sent for every model that matches the document selector, it seems you created 2 of them.

btw `MonacoServices.install` doesn't return a promise

**Reply by KamiC6238** [reply]
> Hello
> 
> It's sent for every model that matches the document selector, it seems you created 2 of them.
> 
> btw `MonacoServices.install` doesn't return a promise

Hi, thanks for your reply.

In my website, a page may have multiple algorithm questions, so there will be multiple editors, and multiple editors use the same language. What should I do to avoid the problem of sending didOpen multiple times in a websocket connection?

**Reply by CGNonofr** [reply]
You want the IntelliSense in one editor and not the other?

You can tweak your document selector using a pattern to only match the model you want

**Reply by KamiC6238** [reply]
> You want the IntelliSense in one editor and not the other?
> 
> You can tweak your document selector using a pattern to only match the model you want

I want the IntelliSense in all of editor, because when one editor created, would create a websocket and send didOpen one times.
But in fact a websocket connection would send didOpen twice time if there has more than one editor and has the same language, that would throw an error on the backend.

![image](https://user-images.githubusercontent.com/23523595/213068305-8a5f50b1-0926-41e7-87e9-990faa9fb661.png)

**Reply by KamiC6238** [reply]
> You want the IntelliSense in one editor and not the other?
> 
> You can tweak your document selector using a pattern to only match the model you want

It works when using a pattern, the `identifier` means the editor identifier, so `monaco-languageclient` can match the model which i wanted.

<img width="297" alt="image" src="https://user-images.githubusercontent.com/23523595/213078405-e1f60af1-225b-45c8-a1fb-a775c803133f.png">

**Reply by CGNonofr** [reply]
Your backend is mimicking a LSP server but can handle only a single open file, is that it?

> It works when using a pattern, the identifier means the editor identifier, so monaco-languageclient can match the model which i wanted.

So problem solved?

**Reply by KamiC6238** [reply]
> Your backend is mimicking a LSP server but can handle only a single open file, is that it?
> 
> > It works when using a pattern, the identifier means the editor identifier, so monaco-languageclient can match the model which i wanted.
> 
> So problem solved?

yes, thanks so much.
